### ISSUE TITLE
Fail to send message

### ISSUE THREAD

**Issue by hwa123** [bitnami/bitnami-docker-kafka#14]
I had a problem when sending message to kafka. The size of message is around 42kb and the kafka-trigger-controller log keeps throwing this error:
```
time="2018-06-11T19:15:26Z" level=error msg="Error: kafka: error while consuming e10ec04419e787d4d056b4f40eac12cc/0: kafka: error decoding packet: unknown magic byte (2)\n"
```
I use default installation for Kafka, no config values being overridden. I'm new to Kafka, so I would be highly appreciated if someone could help me figure out why.

**Comment by tompizmor** [comment]
Hello @hwa123,

I guess that you are able to send simple small messages right? What tool are you using as the client? 
I have found the [following issue](https://github.com/Shopify/sarama/issues/962) in the sarama go client that looks very similar.

**Comment by hwa123** [comment]
@tompizmor Thanks for the reply. I was able to send simple small messages, the tool I'm using as the client is go. I actually followed that link and added `log.message.format.version=0.10.2` inside server.properties. 

Now the kafka-trigger-controller log is showing 
```
time="2018-06-14T18:09:49Z" level=error msg="Failed to send message to function: Error: received error code 413: 413 Payload Too Large"
```
The error as I found is generated from [here](https://github.com/kubeless/kubeless/blob/master/pkg/event-consumers/kafka/kafka-consumer.go) line 84. I believe this could result from limitation from consumer side. I tried setting `fetch.message.max.bytes` to 15MBs inside consumer.properties, also same message size inside producer.properties. However, it still keeps throwing this error.

**Comment by tompizmor** [comment]
Hi! From [kafka documentation](https://kafka.apache.org/documentation/):

> **fetch.message.max.bytes**: The number of bytes of messages to attempt to fetch for each topic-partition in each fetch request. These bytes will be read into memory for each partition, so this helps control the memory used by the consumer. The fetch request size must be at least as large as the maximum message size the server allows or else it is possible for the producer to send messages larger than the consumer can fetch.

So you may also need to tweak others parameters in kafka like `message.max.bytes`. 

Also, if that doesn't help, as you are using kubeless I would recommend to open an issue there as it is possible that other people have run into this too.

**Comment by hwa123** [comment]
Is it possible to add Kafka configuration environment variable that can override this log.message.format.version value inside server.properties? I believe that will be helpful as a workaround to avoid aforementioned error `"kafka: error decoding packet: unknown magic byte (2)\n"`.

**Comment by tompizmor** [comment]
Yes @hwa123, it makes sense. We will work on adding that environment variable. I will update this issue once we release a new version.

**Comment by tompizmor** [comment]
@hwa123 We have released a new revision of the kafka docker image with a new environment variable to configure the log.message.format.version (`KAFKA_LOG_MESSAGE_FORMAT_VERSION`).

This env var is available since `bitnami/kafka:1.1.0-r48`

**Comment by hwa123** [comment]
@tompizmor Thanks, that's great. One more question, I'm getting MessageSizeTooLarge this time caused by Kafka server, which corresponding environment variables do I need to change in order to avoid this error? Since configuration files are no longer persistent after 1.1.0-r41.
